R/tar-terra-vect.R

```diff
@@ -91,93 +92,47 @@ tar_terra_vect <- function(name,
         tidy_eval = tidy_eval
     )
 
-    format <- ifelse(
-        test = filetype == "ESRI Shapefile",
-        #special handling of ESRI shapefiles because the output is a dir of multiple files.
-        yes = create_format_terra_vect_shz(),
-        no =  create_format_terra_vect()
-    )
-
     targets::tar_target_raw(
         name = name,
         command = command,
         pattern = pattern,
         packages = packages,
         library = library,
-        format = format,
+        format = targets::tar_format(
+            read = function(path) {
+                if (filetype == "ESRI Shapefile") {
+                    terra::vect(paste0("/vsizip/{", path, "}"))
+                } else {
+                    terra::vect(path)
+                }
+            },
+            write = function(object, path) {
+                terra::writeVector(
+                    object,
+                    filename = ifelse(filetype == "ESRI Shapefile", paste0(path, ".shz"), path),
+                    filetype = filetype,
+                    overwrite = TRUE,
+                    options = gdal
+                )
+                if (filetype == "ESRI Shapefile") {
+                    file.rename(paste0(path, ".shz"), path)
```

Comment:
```suggestion
                    file.rename(file.path(path, ".shz"), path)
```

Perhaps not needed but it won't hurt

Reply:
Unfortunately that's not going to work:
```
> file.path("path/to/file", ".shz")
[1] "path/to/file/.shz"
```
`fs::path_ext_set` would work, but not worth adding a dependency